### PR TITLE
Add cronjob to drop soft-deleted reviews

### PR DESCRIPTION
**Pull request by fabubaker**

### CODE REVIEW

File: scripts/drop-soft-deleted-reviews-cronjob/cronjob.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        metadata:
+          labels:
+            app: rayyan
+            rayyan.role: cron
+            rayyan.task: drop-soft-deleted-reviews
+        spec:
+          restartPolicy: Never
+          containers:
+          - name: drop-soft-deleted-reviews
+            image: daqcri.azurecr.io/drop-soft-deleted-reviews:1
```
Comment (hammady) [inline scripts/drop-soft-deleted-reviews-cronjob/cronjob.yaml]: Are you going to push this manually? But we need to generate credentials again. What about the public docker hub?

File: scripts/drop-soft-deleted-reviews-cronjob/cronjob.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+            envFrom:
+            - secretRef:
+                name: rayyan-store0-postgres
```
Comment (hammady) [inline scripts/drop-soft-deleted-reviews-cronjob/cronjob.yaml]: You will apply this file twice after changing the secret? You might as well leave a comment for the curious.

File: scripts/drop-soft-deleted-reviews-cronjob/drop-soft-deleted-reviews.sh [hunk 3]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
+
+# Use this script to drop soft deleted reviews from a rayyan-store db.
+# Set DATABASE_URL before running.
+# Set DROP_BATCH_SIZE to adjust number of reviews to drop at a time.
```
Comment (hammady) [inline scripts/drop-soft-deleted-reviews-cronjob/drop-soft-deleted-reviews.sh]: I cannot see this variable defined in the cronjob template.

File: scripts/drop-soft-deleted-reviews-cronjob/cronjob.yaml (hunk 1 above)
Comment (hammady) [reply scripts/drop-soft-deleted-reviews-cronjob/cronjob.yaml]: An alternative way is to run the public postgres image directly, but mount a configmap that contains your script (that you will manually apply as well) and invoke it from the container command.

File: scripts/drop-soft-deleted-reviews-cronjob/drop-soft-deleted-reviews.sh [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+set -e
+
+[ -z "$DATABASE_URL" ] && echo "DATABASE_URL not set! Exiting..." && exit 1;
+
+DROP_BATCH_SIZE=${DROP_BATCH_SIZE:-100}
+echo ">>> Dropping soft deleted reviews with batch size $DROP_BATCH_SIZE."
+
+# Get list of all soft deleted review tables
+psql $DATABASE_URL -c "SELECT table_name FROM information_schema.tables WHERE table_schema='public' AND table_name ~ 'r[0-9]*_[0-9]*$'" \
+| tail -n +3 \
```
Comment (hammady) [inline scripts/drop-soft-deleted-reviews-cronjob/drop-soft-deleted-reviews.sh]: It is more reliable to use `psql` command line switches to get rid of the header. I believe this is `-q` and `-A`.

File: scripts/drop-soft-deleted-reviews-cronjob/drop-soft-deleted-reviews.sh [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+| head -n -2 \
```
Comment (hammady) [inline scripts/drop-soft-deleted-reviews-cronjob/drop-soft-deleted-reviews.sh]: It is hard to predict the output. Can you please put examples?

File: scripts/drop-soft-deleted-reviews-cronjob/drop-soft-deleted-reviews.sh [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+| head -n $DROP_BATCH_SIZE \
+> reviews_to_drop.txt
+
+reviews_to_drop_count=$(cat reviews_to_drop.txt | wc -l)
+
+cat reviews_to_drop.txt \
+| awk '{printf "DROP TABLE %s;\nDROP TABLE %s_customizations_log;\nDROP SEQUENCE %s_customizations_log_id_seq;\n", $1, $1, $1}' \
+| psql $DATABASE_URL --quiet -a $1
+
+echo ">>> Finished. There are $((all_soft_deleted_reviews_count - reviews_to_drop_count)) soft deleted reviews remaining."
```
Comment (hammady) [inline scripts/drop-soft-deleted-reviews-cronjob/drop-soft-deleted-reviews.sh]: Isn't it better and more correct to issue another query to get this number? You can also compare it to the one you are expecting and warn if they differ.

File: scripts/drop-soft-deleted-reviews-cronjob/drop-soft-deleted-reviews.sh (hunk 4 above)
Comment (fabubaker) [reply scripts/drop-soft-deleted-reviews-cronjob/drop-soft-deleted-reviews.sh]: `I tried using `-q`, it doesn't really do anything for headers (just silences notices).

`-A` just turns off alignment which doesn't do much either for headers (it does remove unneeded white-space though). 

Turns out the option is `-t` for removing headers and the footer that gives you the total count of rows.

File: scripts/drop-soft-deleted-reviews-cronjob/cronjob.yaml (hunk 1 above)
Comment (fabubaker) [reply scripts/drop-soft-deleted-reviews-cronjob/cronjob.yaml]: Public docker hub sounds the easiest. I can push it to my personal hub.

Comment (hammady) [reply scripts/drop-soft-deleted-reviews-cronjob/cronjob.yaml]: added you to qcrisw org.

Comment (hammady) [reply scripts/drop-soft-deleted-reviews-cronjob/cronjob.yaml]: Use `rayyanqcri` better.

**Pull request merged**